### PR TITLE
feat: [Tree] Custom leaf icon

### PR DESCRIPTION
**Pull request by xWiiLLz**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [x] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/37143
### 💡 Background and solution

Pretty simple change while preserving the same API for the `showLine` property. Will enable developers to pass a custom leaf icon so that the Tree component can more easily fit various design needs.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Tree `showLeafIcon` could accept ReactNode for a custom leaf icon. |
| 🇨🇳 Chinese | Tree `showLeafIcon` 属性支持自定义图标。 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/tree/Tree.tsx [hunk 1]
```diff
@@ -107,7 +108,7 @@ export interface TreeProps<T extends BasicDataNode = DataNode>
     RcTreeProps<T>,
     'prefixCls' | 'showLine' | 'direction' | 'draggable' | 'icon' | 'switcherIcon'
   > {
-  showLine?: boolean | { showLeafIcon: boolean };
+  showLine?: boolean | { showLeafIcon: boolean; leafIcon?: LeafIcon };
```
Comment (afc163) [inline components/tree/Tree.tsx:111]:
```suggestion
  showLine?: boolean | { showLeafIcon: boolean | React.ReactNode };
```

How about?

Comment (xWiiLLz) [reply components/tree/Tree.tsx:111]: Yeah, good idea. Will make the changes & add tests for the util

Comment (xWiiLLz) [reply components/tree/Tree.tsx]: Updated and added some tests for better coverage 😄 FYI @afc163

File: components/tree/Tree.tsx [hunk 2]
```diff
@@ -11,6 +11,7 @@ import dropIndicatorRender from './utils/dropIndicator';
 import renderSwitcherIcon from './utils/iconUtil';
 
 export type SwitcherIcon = React.ReactNode | ((props: AntTreeNodeProps) => React.ReactNode);
+export type LeafIcon = React.ReactNode | ((props: AntTreeNodeProps) => React.ReactNode);
```
Comment (afc163) [inline components/tree/Tree.tsx:14]:
```suggestion
export type TreeLeafIcon = React.ReactNode | ((props: AntTreeNodeProps) => React.ReactNode);
```

Comment (xWiiLLz) [reply components/tree/Tree.tsx]: Done. Let me know if you find anything else @afc163, otherwise this looks good to merge on my side.

**Comment by xWiiLLz** [comment]
Hey, friendly reminder for this PR. Is there anything else you want me to address before merging @afc163 ? 😄

File: components/tree/index.zh-CN.md [hunk 3]
```diff
@@ -44,7 +44,7 @@ cover: https://gw.alipayobjects.com/zos/alicdn/Xh-oWqg9k/Tree.svg
 | selectable | 是否可选中 | boolean | true |  |
 | selectedKeys | （受控）设置选中的树节点 | string\[] | - |  |
 | showIcon | 是否展示 TreeNode title 前的图标，没有默认样式，如设置为 true，需要自行定义图标相关样式 | boolean | false |  |
-| showLine | 是否展示连接线 | boolean \| {showLeafIcon: boolean} | false |  |
+| showLine | 是否展示连接线 | boolean \| {showLeafIcon: boolean; leafIcon?: ReactNode | ((props: AntTreeNodeProps) => ReactNode)} | false |  |
```
Comment (afc163) [inline components/tree/index.zh-CN.md:47]: `leafIcon` is not existed according to https://github.com/ant-design/ant-design/pull/37144#discussion_r950838350

Comment (xWiiLLz) [reply components/tree/index.zh-CN.md]: Good catch! Just fixed in [#ad19e58](https://github.com/ant-design/ant-design/pull/37144/commits/ad19e58430e2a6811c087dd7df93b190f5b5c8b4)

**Comment by xWiiLLz** [comment]
Up, PR is nearly a month old @afc163

**Pull request merged**